### PR TITLE
[#3012] Fix(build): Fix compile issue after #2866 is merged

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

The proposal changes the task to the function and makes sure the function will not be skipped, also explicitly add the file to jar to make sure it will be executed every time.

### Why are the changes needed?

Fix: #3012 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual test.

### CODE REVIEW

File: common/build.gradle.kts [hunk 1]
```diff
@@ -56,7 +56,7 @@ fun getGitCommitId(): String {
 }
 
 val propertiesFile = "src/main/resources/project.properties"
-val writeProjectPropertiesFile = tasks.register("writeProjectPropertiesFile") {
+fun writeProjectPropertiesFile() {
```
Comment (jerryshao) [inline common/build.gradle.kts:59]: Change to function and add to jar's doFirst to make sure it will run always and cannot skip.

**Comment by jerryshao** [comment]
@coolderli would you please help to check my fix.

**Comment by coolderli** [comment]
> @coolderli would you please help to check my fix.

@jerryshao I try in my local environment. It works. The properties file can be added to the jar successfully.

File: common/build.gradle.kts [hunk 2]
```diff
@@ -85,14 +85,19 @@ val writeProjectPropertiesFile = tasks.register("writeProjectPropertiesFile") {
...
     }
+
+    from("src/main/resources") {
+      include("project.properties").duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+    }
   }
+
   clean {
-    delete("$propertiesFile")
+    delete(file("$propertiesFile").parentFile)
```
[the remark this answers is not in the document]
Comment (coolderli) [reply common/build.gradle.kts:101]: I execute `./gradlew clean` on my computer. And the directory can be deleted successfully.

[the remark this answers is not in the document]
Comment (jerryshao) [reply common/build.gradle.kts:101]: @diqiu50 I referred the original code from server's build script. I think it is more reasonable to delete only one file, not the whole folder. Let me change it back.

**Review by diqiu50 — APPROVED** [review]
LGTM

**Pull request merged**